projects/astc-encoder/astc_load_decompress_image_fuzzer.cc

```diff
@@ -0,0 +1,109 @@
+// Copyright 2020 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "astcenc_internal.h"
+#include "astcenccli_internal.h"
+
+#include <fuzzer/FuzzedDataProvider.h>
+#include <limits>
+#include <vector>
+
+static constexpr uint8_t kUint8Max = std::numeric_limits<uint8_t>::max();
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
+
+  if (size < 19)
```

Comment:
this check depends on the implementation of FuzzedDataProvider, i would not do it, instead later check if you have enough data in the remainder